publisher/internal/analytics/ethereum/operations.go

```diff
@@ -383,3 +340,89 @@ func (op OperationBase) lookupToken(address string) (repository.Token, error) {
 func (op OperationBase) lookupPrice(address string) (repository.TokenPrice, error) {
 	return op.fetchers.priceFetcher.Price(address)
 }
+
+// ----------------------------------------------------------------------
+// --------------- Position methods
+
+func (pos *Position) calculateInterval() {
+
+	lowerRatio := convertTickToRatio(pos.LowerTick, pos.Token0.Decimals, pos.Token1.Decimals)
+	upperRatio := convertTickToRatio(pos.UpperTick, pos.Token0.Decimals, pos.Token1.Decimals)
+
+	if isStableOrNativeInvolved(*pos) && isOrderCorrect(*pos) {
+		lowerRatio = 1 / lowerRatio
+		upperRatio = 1 / upperRatio
+	}
+
+	if lowerRatio > upperRatio {
+		lowerRatio, upperRatio = upperRatio, lowerRatio
+	}
+	pos.LowerRatio, pos.UpperRatio = lowerRatio, upperRatio
+}
+
+func (pos *Position) calculatePosition() {
```

Comment:
Is the `calculateInterval` being called from anywhere else than `calculatePosition`?
If it is not and there are no plans to, you can simply call this function `calculate()`.

Reply:
Will take this into account in the next PR